tortoise/ninja_tortoise.go

```diff
@@ -303,16 +301,16 @@ func (ni *ninjaTortoise) updatePatternTally(newMinGood votingPattern, correction
 	}
 }
 
-func (ni *ninjaTortoise) getCorrEffCounter() (map[types.BlockID]vec, map[types.LayerID]int, func(b *types.BlockHeader)) {
+func (ni *ninjaTortoise) getCorrEffCounter() (map[types.BlockID]vec, map[types.LayerID]int, func(b *types.Block)) {
 	correctionMap := make(map[types.BlockID]vec)
 	effCountMap := make(map[types.LayerID]int)
-	foo := func(b *types.BlockHeader) {
+	foo := func(b *types.Block) {
```

Comment:
I think this is a shame... If we'd put the id in the header we wouldn't need to work with full blocks here.

It's a little more involved to ensure we always have an id, since we need the full block to calculate it, but I think it's worth it. You can look at what we're doing in the ATXs.

Reply:
@noamnelke  its cached leave the tortoise alone